### ISSUE TITLE
Error when attempting to compile new QMK firmware

### ISSUE THREAD

**Issue by arrowtype** [KiserDesigns/AlphaWilly#1]
Hey, I'm loving my new AlphaWilly, but I'm trying to compile new vial firmware so I can:
- Use the latest QMK settings for mod-tap behavior
- Add an additional layer to vial (for fn keys)

I think I've gotten allllmost there, but I'm getting stuck. So far, I have done the following:

1. Pulled the latest from the upstream QMK repo into my local QMK repo (fetch and merge)
2. Run `make git-submodule` to update dependencies
3. Added `keyboards/alphawilly` from the folder in this repo, then added and additional layer to `keyboards/alphawilly/keymaps/vial/keymap.c` ([permalink](https://github.com/arrowtype/qmk_firmware/blob/18eaa0aec758647961809adfcbf13832d9428ac2/keyboards/alphawilly/keymaps/vial/keymap.c))
4. Added a line to config.h based on an error suggestion `#define DYNAMIC_KEYMAP_LAYER_COUNT 5` ([permalink](https://github.com/arrowtype/qmk_firmware/blob/master/keyboards/alphawilly/keymaps/vial/config.h))

But, now I'm getting a more complex error, and I'm hoping someone might be able to suggest what the next step would be. Here's my current build traceback:

```
~/code/qmk_firmware  master ✗                                                                                                                                                                                        6m ⚑  ⍉
▶ qmk compile -kb alphawilly -km vial
Ψ Compiling keymap with gmake --jobs=1 alphawilly:vial


QMK Firmware 0.21.6
⚠ alphawilly: DEVICE_VER in config.h is deprecated in favor of `usb.device_version` in info.json and will be removed at a later date
⚠ alphawilly: MANUFACTURER in config.h is deprecated in favor of `manufacturer` in info.json and will be removed at a later date
⚠ alphawilly: PRODUCT in config.h is deprecated in favor of `keyboard_name` in info.json and will be removed at a later date
⚠ alphawilly: PRODUCT_ID in config.h is deprecated in favor of `usb.pid` in info.json and will be removed at a later date
⚠ alphawilly: VENDOR_ID in config.h is deprecated in favor of `usb.vid` in info.json and will be removed at a later date
☒ alphawilly: No LAYOUTs defined! Need at least one layout defined in info.json.
Making alphawilly with keymap vial

⚠ alphawilly: DEVICE_VER in config.h is deprecated in favor of `usb.device_version` in info.json and will be removed at a later date
⚠ alphawilly: MANUFACTURER in config.h is deprecated in favor of `manufacturer` in info.json and will be removed at a later date
⚠ alphawilly: PRODUCT in config.h is deprecated in favor of `keyboard_name` in info.json and will be removed at a later date
⚠ alphawilly: PRODUCT_ID in config.h is deprecated in favor of `usb.pid` in info.json and will be removed at a later date
⚠ alphawilly: VENDOR_ID in config.h is deprecated in favor of `usb.vid` in info.json and will be removed at a later date
☒ alphawilly: No LAYOUTs defined! Need at least one layout defined in info.json.
arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 8-2019-q3-update) 8.3.1 20190703 (release) [gcc-8-branch revision 273027]
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Compiling: keyboards/alphawilly/alphawilly.c                                                        [OK]
Compiling: .build/obj_alphawilly/src/default_keyboard.c                                             [OK]
Compiling: quantum/keymap_introspection.c                                                           [OK]
Compiling: quantum/quantum.c                                                                        [OK]
Compiling: quantum/bitwise.c                                                                        [OK]
Compiling: quantum/led.c                                                                            [OK]
Compiling: quantum/action.c                                                                         [OK]
Compiling: quantum/action_layer.c                                                                   [OK]
Compiling: quantum/action_tapping.c                                                                 [OK]
Compiling: quantum/action_util.c                                                                    [OK]
Compiling: quantum/eeconfig.c                                                                       [OK]
Compiling: quantum/keyboard.c                                                                       [OK]
Compiling: quantum/keymap_common.c                                                                  [OK]
Compiling: quantum/keycode_config.c                                                                 [OK]
Compiling: quantum/sync_timer.c                                                                     [OK]
Compiling: quantum/logging/debug.c                                                                  [OK]
Compiling: quantum/logging/sendchar.c                                                               [OK]
Compiling: quantum/logging/print.c                                                                  [OK]
Compiling: quantum/bootmagic/bootmagic_lite.c                                                       [OK]
Compiling: quantum/bootmagic/magic.c                                                                [OK]
Compiling: quantum/matrix_common.c                                                                  [OK]
Compiling: quantum/matrix.c                                                                         [OK]
Compiling: quantum/debounce/sym_defer_g.c                                                           [OK]
Compiling: quantum/main.c                                                                           [OK]
Compiling: lib/printf/src/printf/printf.c                                                           [OK]
Compiling: quantum/mousekey.c                                                                       [OK]
Compiling: drivers/eeprom/eeprom_driver.c                                                           [OK]
Compiling: drivers/eeprom/eeprom_wear_leveling.c                                                    [OK]
Compiling: quantum/wear_leveling/wear_leveling.c                                                    [OK]
Compiling: platforms/chibios/drivers/wear_leveling/wear_leveling_rp2040_flash.c                     [OK]
Compiling: quantum/via.c                                                                            [OK]
Compiling: lib/fnv/qmk_fnv_type_validation.c                                                        [OK]
Compiling: lib/fnv/hash_32a.c                                                                       [OK]
Compiling: lib/fnv/hash_64a.c                                                                       [OK]
Compiling: quantum/process_keycode/process_magic.c                                                  [OK]
Compiling: quantum/send_string/send_string.c                                                        [OK]
Compiling: quantum/dynamic_keymap.c                                                                 [OK]
Compiling: quantum/process_keycode/process_grave_esc.c                                              [OK]
Compiling: quantum/process_keycode/process_space_cadet.c                                            [OK]
Compiling: quantum/process_keycode/process_tap_dance.c                                              [OK]
Compiling: quantum/process_keycode/process_tri_layer.c                                              [OK]
Compiling: quantum/tri_layer.c                                                                      [OK]
Assembling: lib/chibios/os/common/startup/ARMCMx/compilers/GCC/crt0_v6m.S                           [OK]
Assembling: lib/chibios/os/common/startup/ARMCMx/compilers/GCC/vectors.S                            [OK]
Assembling: lib/chibios/os/common/ports/ARMv6-M-RP2/compilers/GCC/chcoreasm.S                       [OK]
Compiling: tmk_core/protocol/host.c                                                                 [OK]
Compiling: tmk_core/protocol/report.c                                                               [OK]
Compiling: tmk_core/protocol/usb_device_state.c                                                     [OK]
Compiling: tmk_core/protocol/usb_util.c                                                             [OK]
Compiling: platforms/suspend.c                                                                      [OK]
Compiling: platforms/synchronization_util.c                                                         [OK]
Compiling: platforms/timer.c                                                                        [OK]
Compiling: platforms/chibios/hardware_id.c                                                          [OK]
Compiling: platforms/chibios/platform.c                                                             [OK]
Compiling: platforms/chibios/suspend.c                                                              [OK]
Compiling: platforms/chibios/timer.c                                                                [OK]
Compiling: platforms/chibios/bootloaders/rp2040.c                                                   [OK]
Compiling: tmk_core/protocol/chibios/usb_main.c                                                     [OK]
Compiling: tmk_core/protocol/chibios/chibios.c                                                      [OK]
Compiling: tmk_core/protocol/usb_descriptor.c                                                      In file included from tmk_core/protocol/usb_descriptor.c:42:
tmk_core/protocol/usb_descriptor_common.h:21:29: error: 'LKiser' undeclared here (not in a function)
 #define USBSTR(s) USBCONCAT(L, s)
                             ^
tmk_core/protocol/usb_descriptor_common.h:20:25: note: in definition of macro 'USBCONCAT'
 #define USBCONCAT(a, b) a##b
                         ^
tmk_core/protocol/usb_descriptor.c:1066:42: note: in expansion of macro 'USBSTR'
         .Size                   = sizeof(USBSTR(MANUFACTURER)),
                                          ^~~~~~
./keyboards/alphawilly/config.h:15:31: error: expected ')' before 'Designs'
 #define MANUFACTURER    Kiser Designs
                               ^~~~~~~
tmk_core/protocol/usb_descriptor_common.h:20:28: note: in definition of macro 'USBCONCAT'
 #define USBCONCAT(a, b) a##b
                            ^
tmk_core/protocol/usb_descriptor.c:1066:42: note: in expansion of macro 'USBSTR'
         .Size                   = sizeof(USBSTR(MANUFACTURER)),
                                          ^~~~~~
tmk_core/protocol/usb_descriptor.c:1066:49: note: in expansion of macro 'MANUFACTURER'
         .Size                   = sizeof(USBSTR(MANUFACTURER)),
                                                 ^~~~~~~~~~~~
tmk_core/protocol/usb_descriptor.c:1066:41: note: to match this '('
         .Size                   = sizeof(USBSTR(MANUFACTURER)),
                                         ^
In file included from tmk_core/protocol/usb_descriptor.c:42:
./keyboards/alphawilly/config.h:15:31: error: expected '}' before 'Designs'
 #define MANUFACTURER    Kiser Designs
                               ^~~~~~~
tmk_core/protocol/usb_descriptor_common.h:20:28: note: in definition of macro 'USBCONCAT'
 #define USBCONCAT(a, b) a##b
                            ^
tmk_core/protocol/usb_descriptor.c:1069:35: note: in expansion of macro 'USBSTR'
     .UnicodeString              = USBSTR(MANUFACTURER)
                                   ^~~~~~
tmk_core/protocol/usb_descriptor.c:1069:42: note: in expansion of macro 'MANUFACTURER'
     .UnicodeString              = USBSTR(MANUFACTURER)
                                          ^~~~~~~~~~~~
tmk_core/protocol/usb_descriptor.c:1064:60: note: to match this '{'
 const USB_Descriptor_String_t PROGMEM ManufacturerString = {
                                                            ^
In file included from tmk_core/protocol/usb_descriptor.c:42:
tmk_core/protocol/usb_descriptor_common.h:21:29: error: 'LAlpha' undeclared here (not in a function)
 #define USBSTR(s) USBCONCAT(L, s)
                             ^
tmk_core/protocol/usb_descriptor_common.h:20:25: note: in definition of macro 'USBCONCAT'
 #define USBCONCAT(a, b) a##b
                         ^
tmk_core/protocol/usb_descriptor.c:1074:42: note: in expansion of macro 'USBSTR'
         .Size                   = sizeof(USBSTR(PRODUCT)),
                                          ^~~~~~
./keyboards/alphawilly/config.h:16:31: error: expected ')' before 'Willy'
 #define PRODUCT         Alpha Willy
                               ^~~~~
tmk_core/protocol/usb_descriptor_common.h:20:28: note: in definition of macro 'USBCONCAT'
 #define USBCONCAT(a, b) a##b
                            ^
tmk_core/protocol/usb_descriptor.c:1074:42: note: in expansion of macro 'USBSTR'
         .Size                   = sizeof(USBSTR(PRODUCT)),
                                          ^~~~~~
tmk_core/protocol/usb_descriptor.c:1074:49: note: in expansion of macro 'PRODUCT'
         .Size                   = sizeof(USBSTR(PRODUCT)),
                                                 ^~~~~~~
tmk_core/protocol/usb_descriptor.c:1074:41: note: to match this '('
         .Size                   = sizeof(USBSTR(PRODUCT)),
                                         ^
In file included from tmk_core/protocol/usb_descriptor.c:42:
./keyboards/alphawilly/config.h:16:31: error: expected '}' before 'Willy'
 #define PRODUCT         Alpha Willy
                               ^~~~~
tmk_core/protocol/usb_descriptor_common.h:20:28: note: in definition of macro 'USBCONCAT'
 #define USBCONCAT(a, b) a##b
                            ^
tmk_core/protocol/usb_descriptor.c:1077:35: note: in expansion of macro 'USBSTR'
     .UnicodeString              = USBSTR(PRODUCT)
                                   ^~~~~~
tmk_core/protocol/usb_descriptor.c:1077:42: note: in expansion of macro 'PRODUCT'
     .UnicodeString              = USBSTR(PRODUCT)
                                          ^~~~~~~
tmk_core/protocol/usb_descriptor.c:1072:55: note: to match this '{'
 const USB_Descriptor_String_t PROGMEM ProductString = {
                                                       ^
 [ERRORS]
 | 
 | 
 | 
gmake[1]: *** [builddefs/common_rules.mk:361: .build/obj_alphawilly_vial/usb_descriptor.o] Error 1
gmake: *** [Makefile:392: alphawilly:vial] Error 1
Make finished with errors
```

Thanks so much for any advice or insights on what I might want to do next!

**Comment by arrowtype** [comment]
Oh, the immediate next step seems to be wrapping a couple of strings in quotes:

```
#define MANUFACTURER    "Kiser Designs"
#define PRODUCT         "Alpha Willy"
```

**Comment by arrowtype** [comment]
Okay, I'm a little further, and now stuck on this:

```
~/code/qmk_firmware  master ✗                                                                                                                                                                                        8m ⚑  ⍉
▶ qmk compile -kb alphawilly -km vial
Ψ Compiling keymap with gmake --jobs=1 alphawilly:vial


QMK Firmware 0.21.6
⚠ alphawilly: DEVICE_VER in config.h is deprecated in favor of `usb.device_version` in info.json and will be removed at a later date
⚠ alphawilly: MANUFACTURER in config.h is deprecated in favor of `manufacturer` in info.json and will be removed at a later date
⚠ alphawilly: PRODUCT in config.h is deprecated in favor of `keyboard_name` in info.json and will be removed at a later date
⚠ alphawilly: PRODUCT_ID in config.h is deprecated in favor of `usb.pid` in info.json and will be removed at a later date
⚠ alphawilly: VENDOR_ID in config.h is deprecated in favor of `usb.vid` in info.json and will be removed at a later date
☒ alphawilly: No LAYOUTs defined! Need at least one layout defined in info.json.
Making alphawilly with keymap vial

⚠ alphawilly: DEVICE_VER in config.h is deprecated in favor of `usb.device_version` in info.json and will be removed at a later date
⚠ alphawilly: MANUFACTURER in config.h is deprecated in favor of `manufacturer` in info.json and will be removed at a later date
⚠ alphawilly: PRODUCT in config.h is deprecated in favor of `keyboard_name` in info.json and will be removed at a later date
⚠ alphawilly: PRODUCT_ID in config.h is deprecated in favor of `usb.pid` in info.json and will be removed at a later date
⚠ alphawilly: VENDOR_ID in config.h is deprecated in favor of `usb.vid` in info.json and will be removed at a later date
☒ alphawilly: No LAYOUTs defined! Need at least one layout defined in info.json.
arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 8-2019-q3-update) 8.3.1 20190703 (release) [gcc-8-branch revision 273027]
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Compiling: keyboards/alphawilly/alphawilly.c                                                        [OK]
Compiling: .build/obj_alphawilly/src/default_keyboard.c                                             [OK]
Compiling: quantum/keymap_introspection.c                                                           [OK]
Compiling: quantum/quantum.c                                                                        [OK]
Compiling: quantum/bitwise.c                                                                        [OK]
Compiling: quantum/led.c                                                                            [OK]
Compiling: quantum/action.c                                                                         [OK]
Compiling: quantum/action_layer.c                                                                   [OK]
Compiling: quantum/action_tapping.c                                                                 [OK]
Compiling: quantum/action_util.c                                                                    [OK]
Compiling: quantum/eeconfig.c                                                                       [OK]
Compiling: quantum/keyboard.c                                                                       [OK]
Compiling: quantum/keymap_common.c                                                                  [OK]
Compiling: quantum/keycode_config.c                                                                 [OK]
Compiling: quantum/sync_timer.c                                                                     [OK]
Compiling: quantum/logging/debug.c                                                                  [OK]
Compiling: quantum/logging/sendchar.c                                                               [OK]
Compiling: quantum/logging/print.c                                                                  [OK]
Compiling: quantum/bootmagic/bootmagic_lite.c                                                       [OK]
Compiling: quantum/bootmagic/magic.c                                                                [OK]
Compiling: quantum/matrix_common.c                                                                  [OK]
Compiling: quantum/matrix.c                                                                         [OK]
Compiling: quantum/debounce/sym_defer_g.c                                                           [OK]
Compiling: quantum/main.c                                                                           [OK]
Compiling: lib/printf/src/printf/printf.c                                                           [OK]
Compiling: quantum/mousekey.c                                                                       [OK]
Compiling: drivers/eeprom/eeprom_driver.c                                                           [OK]
Compiling: drivers/eeprom/eeprom_wear_leveling.c                                                    [OK]
Compiling: quantum/wear_leveling/wear_leveling.c                                                    [OK]
Compiling: platforms/chibios/drivers/wear_leveling/wear_leveling_rp2040_flash.c                     [OK]
Compiling: quantum/via.c                                                                            [OK]
Compiling: lib/fnv/qmk_fnv_type_validation.c                                                        [OK]
Compiling: lib/fnv/hash_32a.c                                                                       [OK]
Compiling: lib/fnv/hash_64a.c                                                                       [OK]
Compiling: quantum/process_keycode/process_magic.c                                                  [OK]
Compiling: quantum/send_string/send_string.c                                                        [OK]
Compiling: quantum/dynamic_keymap.c                                                                 [OK]
Compiling: quantum/process_keycode/process_grave_esc.c                                              [OK]
Compiling: quantum/process_keycode/process_space_cadet.c                                            [OK]
Compiling: quantum/process_keycode/process_tap_dance.c                                              [OK]
Compiling: quantum/process_keycode/process_tri_layer.c                                              [OK]
Compiling: quantum/tri_layer.c                                                                      [OK]
Assembling: lib/chibios/os/common/startup/ARMCMx/compilers/GCC/crt0_v6m.S                           [OK]
Assembling: lib/chibios/os/common/startup/ARMCMx/compilers/GCC/vectors.S                            [OK]
Assembling: lib/chibios/os/common/ports/ARMv6-M-RP2/compilers/GCC/chcoreasm.S                       [OK]
Compiling: tmk_core/protocol/host.c                                                                 [OK]
Compiling: tmk_core/protocol/report.c                                                               [OK]
Compiling: tmk_core/protocol/usb_device_state.c                                                     [OK]
Compiling: tmk_core/protocol/usb_util.c                                                             [OK]
Compiling: platforms/suspend.c                                                                      [OK]
Compiling: platforms/synchronization_util.c                                                         [OK]
Compiling: platforms/timer.c                                                                        [OK]
Compiling: platforms/chibios/hardware_id.c                                                          [OK]
Compiling: platforms/chibios/platform.c                                                             [OK]
Compiling: platforms/chibios/suspend.c                                                              [OK]
Compiling: platforms/chibios/timer.c                                                                [OK]
Compiling: platforms/chibios/bootloaders/rp2040.c                                                   [OK]
Compiling: tmk_core/protocol/chibios/usb_main.c                                                     [OK]
Compiling: tmk_core/protocol/chibios/chibios.c                                                      [OK]
Compiling: tmk_core/protocol/usb_descriptor.c                                                       [OK]
Compiling: tmk_core/protocol/chibios/usb_driver.c                                                   [OK]
Compiling: tmk_core/protocol/chibios/usb_util.c                                                     [OK]
Compiling: lib/chibios/os/oslib/src/chmboxes.c                                                      [OK]
Compiling: lib/chibios/os/oslib/src/chmemcore.c                                                     [OK]
Compiling: lib/chibios/os/oslib/src/chmemheaps.c                                                    [OK]
Compiling: lib/chibios/os/oslib/src/chmempools.c                                                    [OK]
Compiling: lib/chibios/os/oslib/src/chpipes.c                                                       [OK]
Compiling: lib/chibios/os/oslib/src/chobjcaches.c                                                   [OK]
Compiling: lib/chibios/os/oslib/src/chdelegates.c                                                   [OK]
Compiling: lib/chibios/os/oslib/src/chfactory.c                                                     [OK]
Compiling: lib/chibios/os/common/startup/ARMCMx/compilers/GCC/crt1.c                                [OK]
Compiling: lib/chibios/os/rt/src/chsys.c                                                            [OK]
Compiling: lib/chibios/os/rt/src/chrfcu.c                                                           [OK]
Compiling: lib/chibios/os/rt/src/chdebug.c                                                          [OK]
Compiling: lib/chibios/os/rt/src/chtrace.c                                                          [OK]
Compiling: lib/chibios/os/rt/src/chvt.c                                                             [OK]
Compiling: lib/chibios/os/rt/src/chschd.c                                                           [OK]
Compiling: lib/chibios/os/rt/src/chinstances.c                                                      [OK]
Compiling: lib/chibios/os/rt/src/chthreads.c                                                        [OK]
Compiling: lib/chibios/os/rt/src/chtm.c                                                             [OK]
Compiling: lib/chibios/os/rt/src/chstats.c                                                          [OK]
Compiling: lib/chibios/os/rt/src/chregistry.c                                                       [OK]
Compiling: lib/chibios/os/rt/src/chsem.c                                                            [OK]
Compiling: lib/chibios/os/rt/src/chmtx.c                                                            [OK]
Compiling: lib/chibios/os/rt/src/chcond.c                                                           [OK]
Compiling: lib/chibios/os/rt/src/chevents.c                                                         [OK]
Compiling: lib/chibios/os/rt/src/chmsg.c                                                            [OK]
Compiling: lib/chibios/os/rt/src/chdynamic.c                                                        [OK]
Compiling: lib/chibios/os/common/ports/ARMv6-M-RP2/chcore.c                                         [OK]
Compiling: lib/chibios/os/hal/osal/rt-nil/osal.c                                                    [OK]
Compiling: lib/chibios/os/hal/src/hal.c                                                             [OK]
Compiling: lib/chibios/os/hal/src/hal_st.c                                                          [OK]
Compiling: lib/chibios/os/hal/src/hal_buffers.c                                                     [OK]
Compiling: lib/chibios/os/hal/src/hal_queues.c                                                      [OK]
Compiling: lib/chibios/os/hal/src/hal_flash.c                                                       [OK]
Compiling: lib/chibios/os/hal/src/hal_mmcsd.c                                                       [OK]
Compiling: lib/chibios/os/hal/src/hal_adc.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_can.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_crypto.c                                                      [OK]
Compiling: lib/chibios/os/hal/src/hal_dac.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_efl.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_gpt.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_i2c.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_i2s.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_icu.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_mac.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_mmc_spi.c                                                     [OK]
Compiling: lib/chibios/os/hal/src/hal_pal.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_pwm.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_rtc.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_sdc.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_serial.c                                                      [OK]
Compiling: lib/chibios/os/hal/src/hal_serial_usb.c                                                  [OK]
Compiling: lib/chibios/os/hal/src/hal_sio.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_spi.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_trng.c                                                        [OK]
Compiling: lib/chibios/os/hal/src/hal_uart.c                                                        [OK]
Compiling: lib/chibios/os/hal/src/hal_usb.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_wdg.c                                                         [OK]
Compiling: lib/chibios/os/hal/src/hal_wspi.c                                                        [OK]
Compiling: lib/chibios/os/hal/ports/common/ARMCMx/nvic.c                                            [OK]
Compiling: lib/chibios/os/hal/ports/RP/RP2040/rp_isr.c                                              [OK]
Compiling: lib/chibios/os/hal/ports/RP/RP2040/hal_lld.c                                             [OK]
Compiling: lib/chibios/os/hal/ports/RP/LLD/DMAv1/rp_dma.c                                           [OK]
Compiling: lib/chibios/os/hal/ports/RP/LLD/GPIOv1/hal_pal_lld.c                                     [OK]
Compiling: lib/chibios/os/hal/ports/RP/LLD/SPIv1/hal_spi_lld.c                                      [OK]
Compiling: lib/chibios/os/hal/ports/RP/LLD/TIMERv1/hal_st_lld.c                                     [OK]
Compiling: lib/chibios/os/hal/ports/RP/LLD/UARTv1/hal_sio_lld.c                                     [OK]
Compiling: lib/chibios/os/hal/ports/RP/LLD/RTCv1/hal_rtc_lld.c                                      [OK]
Compiling: lib/chibios/os/hal/ports/RP/LLD/WDGv1/hal_wdg_lld.c                                      [OK]
Compiling: lib/chibios-contrib/os/hal/ports/RP/LLD/I2Cv1/hal_i2c_lld.c                              [OK]
Compiling: lib/chibios-contrib/os/hal/ports/RP/LLD/PWMv1/hal_pwm_lld.c                              [OK]
Compiling: lib/chibios-contrib/os/hal/ports/RP/LLD/ADCv1/hal_adc_lld.c                              [OK]
Compiling: lib/chibios-contrib/os/hal/ports/RP/LLD/USBDv1/hal_usb_lld.c                             [OK]
Compiling: lib/chibios/os/hal/boards/RP_PICO_RP2040/board.c                                         [OK]
Compiling: lib/chibios/os/hal/lib/streams/chprintf.c                                                [OK]
Compiling: lib/chibios/os/hal/lib/streams/chscanf.c                                                 [OK]
Compiling: lib/chibios/os/hal/lib/streams/memstreams.c                                              [OK]
Compiling: lib/chibios/os/hal/lib/streams/nullstreams.c                                             [OK]
Compiling: lib/chibios/os/hal/lib/streams/bufstreams.c                                              [OK]
Compiling: lib/chibios/os/various/syscalls.c                                                        [OK]
Compiling: platforms/chibios/syscall-fallbacks.c                                                    [OK]
Compiling: platforms/chibios/wait.c                                                                 [OK]
Compiling: platforms/chibios/synchronization_util.c                                                 [OK]
Compiling: lib/pico-sdk/src/rp2_common/hardware_clocks/clocks.c                                     [OK]
Compiling: lib/pico-sdk/src/rp2_common/hardware_pll/pll.c                                           [OK]
Compiling: lib/pico-sdk/src/rp2_common/hardware_pio/pio.c                                           [OK]
Compiling: lib/pico-sdk/src/rp2_common/hardware_timer/timer.c                                       [OK]
Compiling: lib/pico-sdk/src/rp2_common/hardware_flash/flash.c                                       [OK]
Compiling: lib/pico-sdk/src/rp2_common/hardware_gpio/gpio.c                                         [OK]
Compiling: lib/pico-sdk/src/rp2_common/hardware_claim/claim.c                                       [OK]
Compiling: lib/pico-sdk/src/rp2_common/hardware_watchdog/watchdog.c                                 [OK]
Compiling: lib/pico-sdk/src/rp2_common/hardware_xosc/xosc.c                                         [OK]
Compiling: lib/pico-sdk/src/rp2_common/pico_bootrom/bootrom.c                                       [OK]
Compiling: platforms/chibios/vendors/RP/stage2_bootloaders.c                                        [OK]
Compiling: platforms/chibios/vendors/RP/pico_sdk_shims.c                                            [OK]
Assembling: lib/pico-sdk/src/rp2_common/pico_divider/divider.S                                      [OK]
Assembling: lib/pico-sdk/src/rp2_common/pico_int64_ops/pico_int64_ops_aeabi.S                       [OK]
Linking: .build/alphawilly_vial.elf                                                                 [ERRORS]
 | 
 | /opt/homebrew/Cellar/arm-gcc-bin@8/8-2019-q3-update_3/bin/../lib/gcc/arm-none-eabi/8.3.1/../../../../arm-none-eabi/bin/ld: .build/obj_alphawilly_vial/quantum/process_keycode/process_tap_dance.o: in function `preprocess_tap_dance':
 | /Users/stephennixon/code/qmk_firmware/quantum/process_keycode/process_tap_dance.c:121: undefined reference to `tap_dance_actions'
 | /opt/homebrew/Cellar/arm-gcc-bin@8/8-2019-q3-update_3/bin/../lib/gcc/arm-none-eabi/8.3.1/../../../../arm-none-eabi/bin/ld: .build/obj_alphawilly_vial/quantum/process_keycode/process_tap_dance.o: in function `process_tap_dance':
 | /Users/stephennixon/code/qmk_firmware/quantum/process_keycode/process_tap_dance.c:155: undefined reference to `tap_dance_actions'
 | /opt/homebrew/Cellar/arm-gcc-bin@8/8-2019-q3-update_3/bin/../lib/gcc/arm-none-eabi/8.3.1/../../../../arm-none-eabi/bin/ld: .build/obj_alphawilly_vial/quantum/process_keycode/process_tap_dance.o: in function `tap_dance_task':
 | /Users/stephennixon/code/qmk_firmware/quantum/process_keycode/process_tap_dance.c:174: undefined reference to `tap_dance_actions'
 | collect2: error: ld returned 1 exit status
 | 
gmake[1]: *** [builddefs/common_rules.mk:268: .build/alphawilly_vial.elf] Error 1
gmake: *** [Makefile:392: alphawilly:vial] Error 1
Make finished with errors
```

**Comment by arrowtype** [comment]
Okay, in the end, I followed these steps:
- Most importantly, I had to git clone [https://github.com/vial-kb/vial-qmk](https://github.com/vial-kb/vial-qmk) and work from in there, rather than from my qmk-firmware repo.
- I updated the submodules of my new vial-qmk repo with `make git-submodules`
- I changed a couple of things in the alphawilly info.json and vial keymaps.... current settings are at https://github.com/arrowtype/vial-qmk/tree/ef3afa679534a86b71792dc6d20e5ac4d1841951/keyboards/alphawilly
- I built the new vial firmware with `qmk compile -kb alphawilly -km vial`. This built a `.uf2` file in the `.build` directory
- In Vial, I then assigned a Reset key combo, and saved my layout again
- I reset the board
- I copied the new `.uf2` file into the RPI-RP2 volume on the command line (because Finder doesn't allow it), with `cp -X /Users/stephennixon/code/vial-qmk/.build/alphawilly_vial.uf2 /Volumes/RPI-RP2` (the path to the `.uf2` file will obviously vary for others)

And it works! I now have an extra layer, which I have used for all my Fn keys.

Thanks very much to @KiserDesigns for helping me out on the 40% keyboards Discord!
